plugins/inputs/tail/tail.go

```diff
@@ -86,89 +91,88 @@ func (t *Tail) Start(acc telegraf.Accumulator) error {
 	defer t.Unlock()
 
 	t.acc = acc
-	t.tailers = make(map[string]*tail.Tail)
-
-	return t.tailNewFiles(t.FromBeginning)
-}
-
-func (t *Tail) tailNewFiles(fromBeginning bool) error {
-	var seek *tail.SeekInfo
-	if !t.Pipe && !fromBeginning {
-		seek = &tail.SeekInfo{
-			Whence: 2,
-			Offset: 0,
-		}
-	}
+	t.lines = make(chan *logline.LogLine)
+	t.files = make(map[string]bool)
 
 	var poll bool
 	if t.WatchMethod == "poll" {
 		poll = true
 	}
 
-	// Create a "tailer" for each file
+	var err error
+	t.watcher, err = watcher.NewLogWatcher(t.PollInterval.Duration, !poll, watcher.Logger(taillog.DiscardingLogger))
+	if err != nil {
+		defer close(t.lines)
+		return err
+	}
+
+	opts := []tailer.Option{tailer.Logger(taillog.DiscardingLogger)}
+	if t.FromBeginning {
+		opts = append(opts, tailer.OneShot)
+	}
+
+	t.tailer, err = tailer.New(t.lines, t.watcher, opts...)
+	if err != nil {
+		defer t.watcher.Close()
+		defer close(t.lines)
+		return err
+	}
+
+	parser, err := t.parserFunc()
+	if err != nil {
+		defer t.tailer.Close()
```

Comment:
Do we need to also close lines/watcher?

Reply:
t.tailer.Close() will close lines and watcher; I'll add comments noting this.